src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -313,97 +312,56 @@ public void setPeriod(String period) {
-     */
-    private static boolean isInlineTagPresent(DetailNode ast) {
-        return ast.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG
-                || ast.getType() == JavadocTokenTypes.HTML_ELEMENT
-                && getInlineTagNodeWithinHtmlElement(ast) != null;
+    @Override
+    public void leaveJavadocToken(DetailNode ast) {
+        if (ast.getType() == JavadocTokenTypes.JAVADOC
+                && !startsWithInheritDoc(ast)) {
+            if (shouldCheckJavadocToken) {
+                checkJavadocToken(ast);
+            }
+            shouldCheckJavadocToken = true;
+        }
     }
 
     /**
-     * Returns an inline javadoc tag node that is within a html tag.
+     * Check node of type {@link JavadocTokenTypes#JAVADOC}.
      *
-     * @param ast html tag node.
-     * @return inline summary javadoc tag node or null if no node is found.
+     * @param ast ast
      */
-    private static DetailNode getInlineTagNodeWithinHtmlElement(DetailNode ast) {
-        DetailNode node = ast;
-        DetailNode result = null;
-        // node can never be null as this method is called when there is a HTML_ELEMENT
-        if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
-            result = node;
-        }
-        else if (node.getType() == JavadocTokenTypes.HTML_TAG) {
-            // HTML_TAG always has more than 2 children.
-            node = node.getChildren()[1];
-            result = getInlineTagNodeWithinHtmlElement(node);
+    private void checkJavadocToken(DetailNode ast) {
+        final String summaryDoc = getSummarySentence(ast);
+        if (summaryDoc.isEmpty()) {
+            log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
         }
-        else if (node.getType() == JavadocTokenTypes.HTML_ELEMENT
-                // Condition for SINGLETON html element which cannot contain summary node
-                && node.getChildren()[0].getChildren().length > 1) {
-            // Html elements have one tested tag before actual content inside it
-            node = node.getChildren()[0].getChildren()[1];
-            result = getInlineTagNodeWithinHtmlElement(node);
+        else if (!period.isEmpty()) {
+            final String firstSentence = getFirstSentence(ast);
+            final int endOfSentence = firstSentence.lastIndexOf(period);
+            if (!summaryDoc.contains(period)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_FIRST_SENTENCE);
+            }
+            if (endOfSentence != -1
+                    && containsForbiddenFragment(
+                            firstSentence.substring(0, endOfSentence))) {
+                log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC);
+            }
         }
-        return result;
     }
 
     /**
      * Checks if the first tag inside ast is summary tag.
      *
-     * @param javadoc root node.
+     * @param inlineTag root node.
      * @return {@code true} if first tag is summary tag.
```

Comment:
whole javadoc description is misleading now, there is no "first" tags

Reply:
Done